src/main/java/seedu/medmoriser/logic/commands/QuizCommand.java

```diff
@@ -0,0 +1,65 @@
+package seedu.medmoriser.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import javafx.collections.ObservableList;
+import seedu.medmoriser.logic.commands.exceptions.CommandException;
+import seedu.medmoriser.model.Model;
+import seedu.medmoriser.model.qanda.QAndA;
+import seedu.medmoriser.model.qanda.QuestionContainsKeywordsPredicate;
+import seedu.medmoriser.model.qanda.TagContainsKeywordsPredicate;
+import seedu.medmoriser.model.qanda.QAndAContainsKeywordsPredicate;
+
+import java.util.List;
+import java.util.Random;
+import java.util.function.Predicate;
+
+/**
+ * Starts a new Quiz for the user.
+ */
+public class QuizCommand extends Command {
+
+    public static boolean isQuiz = false;
```

Comment:
I'm not sure if the isQuiz state should be placed in the command. Do you think it would be possible to put it under a new class that controls the system's state? @jonfoocy what do you think about this?

Reply:
I'm not very sure about this either, I just added a getter and setter for isQuiz and made it a private static boolean.

Reply:
Good point actually I did the same thing for my implementation of the ListCommand. Maybe we can extract it and put it in the UiManager component?